physics/archimedes_principle.py

```diff
@@ -30,13 +30,27 @@ def archimedes_principle(
     4885.3
     >>> archimedes_principle(fluid_density=997, volume=0.7)
     6844.061035
+    >>> archimedes_principle(fluid_density=997, volume=-0.7)
+    Traceback (most recent call last):
+        ...
+    ValueError: Impossible Object volume
+    >>> archimedes_principle(fluid_density=0, volume=0.7)
+    Traceback (most recent call last):
+        ...
+    ValueError: Impossible fluid density
+    >>> archimedes_principle(fluid_density=997, volume=0.7, gravity=0)
+    0.0
+    >>> archimedes_principle(fluid_density=997, volume=0.7, gravity=-9.8)
+    Traceback (most recent call last):
+        ...
+    ValueError: Impossible Gravity
     """
 
     if fluid_density <= 0:
         raise ValueError("Impossible fluid density")
-    if volume < 0:
+    if volume <= 0:
         raise ValueError("Impossible Object volume")
-    if gravity <= 0:
+    if gravity < 0:
```

Comment:
Please revert these changes.  It is useful to be able to make theoretical calculations even if we are unable to reproduce those conditions.

Reply:
should I remove the tests as well or just make the signs as they were before?

Reply:
The tests need to follow the code.  It is OK to have the tests but their output must pass.

Reply:
The numerical tests are passing. Just added some more tests to see for when the inputs are invalid physically as in negative gravity, non positive volume and non positive density as they were already in the original code by the author.

Reply:
please check @cclauss

Reply:
Dude!  There are 215 pull requests open.  Perhaps we will get to review this one.